File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.swift
import Foundation

func solution(_ s:String) -> [Int] {
    
    var result : [Int] = []
    var dic : [String:Int] = [:]

    
    for (a,i) in s.enumerated() {
        if dic[String(i)] != nil{
            result.append(a - dic[String(i)]!)
            dic[String(i)] = a
            
        }else{
            result.append(-1)
            dic[String(i)] = a
        }
        
    }
    
    return result
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const result = []
    const dic = {}
    for(let i=0; i<s.length; i++) {
        if(dic[s[i]] === undefined) {
            result.push(-1)
        } else {
            result.push(i - dic[s[i]]) 
        }
        dic[s[i]] = i
    }
    return result
}